### PR TITLE
feat: RFC section ~ #26

### LINKED ISSUE #26: Create RFC Section
* [x] Create an `/RFCs` directory
* [x] Add an `/RFCs/template.md` document
* [x] Add an `/RFCs/README.md` that explains the process

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
Closes https://github.com/fission-codes/spec/issues/26. cc @expede

* [X] Create an `/RFCs` directory
* [X] Add an `/RFCs/template.md` document
* [X] Add an `/RFCs/README.md` that explains the process

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*~
+
+# Mac files
+.DS_Store
```
Comment (expede) [inline .gitignore:4]: Works for me! I used to add a bunch of stuff like this to repos, but then (IIRC) Steven pointed out that we should probably put this stuff in our `~/.config/git/ignore`s. I'm happy either way though 👍

File: rfcs/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+```
+|_ spec
+  |_ rfcs
+    |_ Why-Chocolate-Cake-Is-Best
+      |_ RFC.md
+      |_ diagrams
+        |_ cake.png
+    |_ README.md
+    |_ template.md
+```
```
Comment (expede) [inline rfcs/README.md]:
```suggestion
└── spec
     ├── rfcs
     │    ├── Why-Chocolate-Cake-Is-Best
     │    ├── RFC.md
     │    └── diagrams
     │         └── cake.png
     ├── README.md
     └── template.md
```

File: rfcs/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+- `Suspended`: during the RFC process, the engaged parties determined that
+  though the proposal might be a good idea, it's not the right time for it. The
+  PR can be marked `Suspended` and returned to later.
+- `Rejected`: a consensus was reached that this was not the right solution for
+  the problem. The associated PR will be marked `Rejected`.
+
+During the `Draft` and `Discussion` stages of an RFC, it may be helpful to write
+prototypes or gists to better convey a line of thinking.
+
+Once an RFC merged into `main`, it's considered `Published`.
```
Comment (expede) [inline rfcs/README.md]: I'm less certain about this section for our uses. I see this a lot in community projects, but today this repo is mostly used for agreeing on how something work and we hammer out stuff in PR. That's not to say "no", but I mainly don't think we'll just not merge (or delete) rejected proposals

File: rfcs/README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+### On Timing and Interaction
+
+Once your RFC is ready for discussion, engage with and notify any interested
+parties over discord, etc. **Sell what you're building**. If an RFC sits in
+`Draft` or `Discussion` for more than 30 days without feedback or clarity, it
+will be updated with an `Abandoned` label and the PR will be closed until the
+author(s) picks it back up again.
```
Comment (expede) [inline rfcs/README.md]: Is there a bot that we could use for this? IIRC the EIPs repo had one.

File: rfcs/template.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+authors:
+
+- `[Author1](Github Link for Author)`
+
+necessary reviewers:
+
+- [ ] Reviewer 1
+- [ ] Reviewer 2
+- [ ] Reviewer 3
```
Comment (expede) [inline rfcs/template.md]: Does this differ from the "request review" feature and/or "review required by codeowners" feature in GH?

File: rfcs/template.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+### Introduction
+
+One paragraph explanation of the feature(s), change(s), or addition(s) being
+proposed. A diagram can often be helpful here.
```
Comment (expede) [inline rfcs/template.md:10]:
```suggestion
<!--
One paragraph explanation of the feature(s), change(s), or addition(s) being
proposed. A diagram can often be helpful here.
-->
```

**Review by expede — COMMENTED** [review]
@zeeshanlakhani overall I like it! Just a couple of questions & comments. It's actually not that far off from our current copy/paste format in UCAN &c, so I'm all for having it in a template 👍

File: rfcs/README.md (hunk 4 above)
Comment (zeeshanlakhani) [reply rfcs/README.md]: @expede probably. Plan to look into bots for numbering and these kinda things at some point.

File: rfcs/template.md (hunk 5 above)
Comment (zeeshanlakhani) [reply rfcs/template.md]: Yeah. Good question. It's not that codeowners are always the stakeholders per see, but if we use the PR request review correctly, we can remove this. I'll make this a note instead.

File: .gitignore (hunk 1 above)
Comment (zeeshanlakhani) [reply .gitignore:4]: yeah. I have that too locally, but can't always force people. These two bits are normcore for most working in text repos.

File: rfcs/README.md (hunk 3 above)
Comment (zeeshanlakhani) [reply rfcs/README.md]: makes sense. I'll remove the rejected bit.

File: rfcs/README.md (hunk 4 above)
Comment (zeeshanlakhani) [reply rfcs/README.md]: I'll open a couple issues for these in general.

File: rfcs/template.md (hunk 6 above)
Comment (zeeshanlakhani) [reply rfcs/template.md:10]: Don't think we want this unrendered in the markdown right?

File: rfcs/README.md (hunk 3 above)
Comment (zeeshanlakhani) [reply rfcs/README.md]: The draft/discussion stuff is obvious for sure, but it's good to just jot down for clarity (esp for new peeps joining).

File: rfcs/template.md (hunk 6 above)
Comment (expede) [reply rfcs/template.md:10]: Or maybe another way: why _do_ we want this rendered in markdown? It's certainly not an over my dead body, but having this available in your editor but not render if you forget to pull it out of the doc seems useful. But just a suggestion!

Comment (zeeshanlakhani) [reply rfcs/template.md:10]: Ah. I guess I look at it for those reading the template on GH (not yet having pulled it down).

File: rfcs/README.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
+# RFCs at Fission
+
+This introduction should provide you everything you need to get you started in
+the wonderful world of RFCs at Fission. Read on to discover how our organization
+handles proposals for additions and changes to the p# RFCs at Fission
```
Comment (expede) [inline rfcs/README.md]: Is `p#` a convention that I'm not familiar with?

Comment (zeeshanlakhani) [reply rfcs/README.md]: that's just a typo :).

Comment (zeeshanlakhani) [reply rfcs/README.md]: fixed in next push.

**Review by expede — APPROVED** [review]
Looks good to me!

**Pull request merged**